shelf/endpoint_decorators.py

```diff
@@ -214,18 +218,20 @@ def decode_request(self, container):
             Returns:
                 object | None: decoded JSON from request. None if invalid.
         """
-        data = container.request.get_json(silent=True, force=True)
+        if container.request.data:
+            data = container.request.get_json(silent=True, force=True)
 
-        if not isinstance(data, (list, dict)):
-            container.context.add_error(ErrorCode.INVALID_REQUEST_DATA_FORMAT)
-            data = None
+            if not isinstance(data, (list, dict)):
+                container.context.add_error(ErrorCode.INVALID_REQUEST_DATA_FORMAT)
+                data = None
+        else:
+            data = deepcopy(default_request_body)
```

Comment:
Python passes objects by reference, so we need to do a deep copy so it won't assign `data` to the reference of `default_request_body`. I can add this to the doc block in the function too.